map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,201 @@
+        StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline routePolyline = (Polyline) Polyline.fromJson(polylineData,
+        SpatialReference.create(3857));
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        routePolyline, new SimulationParameters(Calendar.getInstance(), 30.0, 0.0, 0.0));
+
+      // start the simulated location data source
+      simulatedLocationDataSource.startAsync();
+      simulatedLocationDataSource.addLocationChangedListener(locationChangedEvent -> {
+        // if location tracking is turned off, do not add to the polyline
+        if (!isTrackingEnabled) {
+          return;
+        }
+        // get the position as a point from locationChangedEvent
+        Point position = locationChangedEvent.getLocation().getPosition();
+
+        // add the new point to the polyline
+        polylineBuilder.addPoint(position);
+
+        // add the new point to the graphics overlay
+        locationHistoryOverlay.getGraphics().add(new Graphic(position));
+
+        // reset the old polyline connecting the points
+        locationHistoryLineOverlay.getGraphics().clear();
+
+        // add the updated polyline to the graphics overlay
+        locationHistoryLineOverlay.getGraphics().add(new Graphic(polylineBuilder.toGeometry()));
+      });
+
+      // configure the map view's location display to follow the simulated location data source
+      mapView.getLocationDisplay().setLocationDataSource(simulatedLocationDataSource);
+      mapView.getLocationDisplay().setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      mapView.getLocationDisplay().setInitialZoomScale(7000);
+
+      trackingButton.setOnAction(event -> {
+        // if the user has panned away from the location display, turn it on again
+        if (mapView.getLocationDisplay().getAutoPanMode() == LocationDisplay.AutoPanMode.OFF) {
+          mapView.getLocationDisplay().setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+        }
+        // toggle the location tracking when the button is clicked
+        if (isTrackingEnabled) {
+          trackingButton.setText("Start Tracking");
+          isTrackingEnabled = false;
+        } else {
+          trackingButton.setText("Stop Tracking");
+          isTrackingEnabled = true;
+        }
+      });
+
+      // add the map view and tracking button to the stack pane
+      stackPane.getChildren().addAll(mapView, trackingButton);
+      StackPane.setAlignment(trackingButton, Pos.TOP_LEFT);
+      StackPane.setMargin(trackingButton, new Insets(10, 0, 0, 10));
+    } catch (Exception e) {
```

Comment:
I think a newline here would be good to separate this from the block above
```suggestion
    
    } catch (Exception e) {
```